src/scheduler/node_info.cpp

```diff
@@ -2368,9 +2380,13 @@ eval_selspec(status *policy, selspec *spec, place *placespec,
 	} else
 		clear_schd_error(failerr);
 
-	/* clear the node scratch space for use for node searching */
+	/* Remove visited, scattered and ineligible bits from ncsr for node searching
+	 * Since statebusy is a per cycle bit, it shouldn't be changed
+	 */
 	for (i = 0; ninfo_arr[i] != NULL; i++) {
-		ninfo_arr[i]->nscr = NSCR_NONE;
+		ninfo_arr[i]->nscr &= ~NSCR_VISITED;
+		ninfo_arr[i]->nscr &= ~NSCR_SCATTERED;
+		ninfo_arr[i]->nscr &= ~NSCR_INELIGIBLE;
```

Comment:
Combine this into one line: ninfo_arr[i]->nscr &= (NSCR_VISITED | NSCR_SCATTERED | NSCR_INELIGIBLE);